components/plans/create/AttributePicker.tsx

```diff
@@ -0,0 +1,231 @@
+import { PlusIcon, XMarkIcon } from "@heroicons/react/20/solid";
+import clsx from "clsx";
+import { useState } from "react";
+import { useMap } from "react-map-gl";
+import Button from "../../Button";
+import CardSlider from "../../CardSlider";
+
+interface AttributesPickerProps {
+  attributes: string[] | undefined;
+  selectedAttributes: string[][];
+  setSelectedAttributes: (stops: string[][]) => void;
+  attributesIndex: number;
+  setAttributesIndex: (index: number) => void;
+  setSelectedTabIndex: (index: number) => void;
+  suggestions: Venue[] | undefined;
+}
+
+export default function AttributesPicker({
+  attributes,
+  selectedAttributes,
+  setSelectedAttributes,
+  attributesIndex,
+  setAttributesIndex,
+  setSelectedTabIndex,
+  suggestions,
+}: AttributesPickerProps) {
+  const { map } = useMap();
+
+  const addStop = (stopIndex: number) => {
+    if (selectedAttributes[stopIndex - 1].length > 0) {
+      let stops = selectedAttributes;
+      stops[attributesIndex] = [];
+      setSelectedAttributes([...stops]);
+      setAttributesIndex(stopIndex + 1);
+    }
+  };
+
+  const removeStop = (stopIndex: number) => {
+    let stops = selectedAttributes;
+    if (attributesIndex === 1) return;
+
+    stops?.splice(stopIndex, 1);
+    setSelectedAttributes([...stops]);
+    setAttributesIndex(attributesIndex - 1);
+  };
+
+  const removeAttribute = (stopIndex: number, attribute: string) => {
+    if (selectedAttributes === undefined) return;
+    let stops = selectedAttributes;
+    const attributes = stops[stopIndex];
+
+    const attributeIndex = attributes.indexOf(attribute);
+    attributes.splice(attributeIndex, 1);
+    stops[stopIndex] = attributes;
+    setSelectedAttributes([...stops]);
+
+    if (attributes.length === 0 && selectedAttributes.length - 1 !== stopIndex)
+      return removeStop(stopIndex);
+  };
+
+  const addAttribute = (currentIndex: number, attribute: string) => {
+    let stopAttributes;
+
+    let stops = selectedAttributes;
+    stopAttributes = selectedAttributes[currentIndex];
+
+    if (!stopAttributes.includes(attribute)) stopAttributes.push(attribute);
+
+    stops[currentIndex] = stopAttributes;
+    setSelectedAttributes([...stops]);
+  };
+
+  const attributesSelected = () =>
```

Comment:
this doesn't need to be a function, it then reads a little better inline when used in the JSX. Also consider the language isAttributesSelected.